test/test-readability.js

```diff
@@ -47,11 +48,44 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
   };
 });
 
+function suite(result, expectedContent, expectedMetadata) {
+  it("should return a result object", function() {
+    expect(result).to.include.keys("content", "title", "excerpt", "byline");
+  });
+
+  it("should extract expected content with jsdom", function() {
+    expect(expectedContent).eql(prettyPrint(result.content));
+  });
+
+  it("should extract expected title", function() {
+    expect(expectedMetadata.title).eql(result.title);
+  });
+
+  it("should extract expected byline", function() {
+    expect(expectedMetadata.byline).eql(result.byline);
+  });
+
+  it("should extract expected excerpt", function() {
+    expect(expectedMetadata.excerpt).eql(result.excerpt);
+  });
+}
+
+function removeCommentNodesRecursively(node) {
+  for (var n = 0, length = node.childNodes.length; n < length; n++) {
+    var child = node.childNodes[n];
+    if (!child) continue; // JSDOM is dumb sometimes :/
```

Comment:
> (please note I'm not fascinated with the idea of investigating this further atm, but maybe you are?).

I am. And it was stupid of me asking to be told where to begin. I should just comment your line, notice the bug in the tests and start from there.
I am interested, because jsdom is the closest attempt I'm aware of of a conformant standalone DOM in Node.js. 

> But you're invited to fork and maintain your own version if you will (hint: don't).

:-p I'll try to repro the bug you found, find out whether it's still in 4.x and report to them if so.